ios/Classes/SwiftOptimoveFlutterSdkPlugin.swift

```diff
@@ -0,0 +1,366 @@
+        let id = (call.arguments as! Dictionary<String, Any>)["id"] as! Int
+        
+        let inboxItems: [InAppInboxItem] = OptimoveInApp.getInboxItems()
+        var deleted = false
+        inboxItems.forEach { item in
+            if item.id == id {
+                deleted = OptimoveInApp.deleteMessageFromInbox(item: item)
+            }
+        }
+        
+        result(deleted)
+    }
+    
+    fileprivate func handleInAppInboxSummary(_ result: @escaping FlutterResult) {
+        OptimoveInApp.getInboxSummaryAsync(inboxSummaryBlock: { inAppInboxSummary in
+            var inAppInboxSummaryMap = [String: Any?]()
+            inAppInboxSummaryMap["totalCount"] = inAppInboxSummary?.totalCount
+            inAppInboxSummaryMap["unreadCount"] = inAppInboxSummary?.unreadCount
+            result(inAppInboxSummaryMap)
+        })
+    }
+    
+    
+    fileprivate func emitPushNotificationReceivedEvent(pushNotification: PushNotification){
+        var notificationMap = [String: Any?]()
+        notificationMap["type"] = EventTypes.pushReceived.rawValue
+        notificationMap["data"] = getPushNotificatioMap(from: pushNotification)
+        self.eventSink?(notificationMap)
+    }
+    
+    fileprivate func emitPushNotificationOpenedEvent(pushNotification: PushNotification){
+        var notificationMap = [String: Any?]()
+        notificationMap["type"] = EventTypes.pushOpened.rawValue
+        notificationMap["data"] = getPushNotificatioMap(from: pushNotification)
+        self.eventSink?(notificationMap)
+    }
+    
+    fileprivate func emitDeeplinkResolved(deepLinkResolution: DeepLinkResolution){
+        var deeplinkResolvedMap = [String: Any?]()
+        deeplinkResolvedMap["type"] = EventTypes.pushDeepLinkResolved.rawValue
+        
+        var data = [String: Any?]()
+        var urlString: String
+        
+        switch deepLinkResolution {
+        case .lookupFailed(let dl):
+            urlString = dl.absoluteString
+            data["resolution"] = 0
+        case .linkNotFound(let dl):
+            urlString = dl.absoluteString
+            data["resolution"] = 1
+        case .linkExpired(let dl):
+            urlString = dl.absoluteString
+            data["resolution"] = 2
+        case .linkLimitExceeded(let dl):
+            urlString = dl.absoluteString
+            data["resolution"] = 3
+        case .linkMatched(let dl):
+            urlString = dl.url.absoluteString
+            data["resolution"] = 4
+            var content = [String: Any?]()
+            content["title"] = dl.content.title
+            content["description"] = dl.content.description
+            
+            var link = [String: Any?]()
+            link["content"] = content
+            link["data"] = dl.data
+            
+            data["link"] = link
+        }
+        data["url"] = urlString
+        deeplinkResolvedMap["data"] = data
+        self.eventSink?(deeplinkResolvedMap)
```

Comment:
Similar Q, do we need to queue the event if listener is not ready yet for cold starts?

Reply:
Changed to a queue